src/fairseq2/models/loader.py

```diff
@@ -356,7 +358,18 @@ def __call__(
         if isinstance(model_name_or_card, AssetCard):
             card = model_name_or_card
         else:
-            card = self._asset_store.retrieve_card(model_name_or_card)
+            try:
+                card = self._asset_store.retrieve_card(model_name_or_card)
+            except AssetNotFoundError as err:
+                model_path = Path(model_name_or_card)
+                # If the card is not found, try looking it up by interpreting model_name as a path to the yaml card.
+                if model_path.exists() and model_path.suffix == ".yaml":
+                    import yaml
+                    with open(model_path, "r", encoding="utf-8") as f:
+                        card_data = yaml.full_load(f)
```

Comment:
I would prefer this to be `yaml.safe_load()` instead since we take an arbitrary path. Is there a specific requirement for `full_load()`?